pkgs/servers/search/sonic-server/default.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, rustPlatform
+, llvmPackages
+, clang
+, fetchFromGitHub
+, withJemallocAllocator ? true
+, withChineseTokenizer ? true
+, ...
```

Comment:
```suggestion
```
Why would you want to turn them off? They don't seem to increase closure size.

Reply:
this is a more convenient and intuitive way of configuring the package when we support features through parameters for `pkgs.sonic-server.override {}` than through `pkgs.sonic-server.overrideAttrs (oldAttrs: { })`

Reply:
Yes, I know but why do we want them. Wouldn't it be even better if we enabled those features all the time?

Reply:
I haven't used these features on my server for three years. Maybe someone else doesn't need them either. But they are default features in sonic-server. So I turned them on by default to be expected.

I can remove them as you suggested and I will use overrideAttrs. but I don't understand where the boundary is in the parameters. Why in some cases we can do that and in some cases we can't.

Reply:
> I haven't used these features on my server for three years.

but what happens if those features are unused? If they just sit there and do nothing and have no significant impact on the closure size than having those options here is pointless.

>  but I don't understand where the boundary is in the parameters. Why in some cases we can do that and in some cases we can't.

Ideally we want as little options as possible. Especially options which do not require extra dependencies and have little to no impact on the final output size are not valuable to add unless they disable features which have significant security impact.

Reply:
I've removed these parameters as you suggested